### PR TITLE
Update docs

### PR DESCRIPTION
**Pull request by markcmiller86**
- I wanted to capture some of the nitty gritty details about handling of `cd_values` in the documentation.
- Fix links to HDF5 docs to link to newer HDF5 documentation instead of stuff in `support.hdfgroup...`
- Fix links to ZFP pages
- Add info about partial I/O requests combined with chunking
- Update remarks about endianness

### CODE REVIEW

**Review by brtnfld — APPROVED** [review]
lgtm, I made some minor edits.

File: docs/cd_vals.rst [hunk 1]
```diff
@@ -12,24 +12,26 @@ The second use is in *persisting* auxiliary data for a filter to the dataset's o
...
+Setting aside the possibility of differing floating point formats between the producer and consumers, any endianness handling the HDF5_ library does for the 4-byte entries in ``cd_values`` will certainly not ensure proper endianness handling of larger values.
 It is impossible for command-line tools like ``h5ls`` and ``h5dump`` to display such data correctly.
 
-Fortunately, the ZFP library has already been designed to handle these issues as part of the ZFP's *native* stream header.
-Consequently, the H5Z-ZFP filter uses the ``cd_values`` that is persisted to a dataset's object header to store ZFP's stream header.
-ZFP's stream header is stored starting at ``&cd_values[1]``. 
-``cd_values[0]`` is used to stored H5Z-ZFP filter and ZFP library and ZFP encoder version information.
+Fortunately, the ZFP_ library has already been designed to handle these issues as part of the ZFP_'s *native* stream header.
+But, the ZFP_ library handles these issues in an endian-agnostic way. 
+This has the effect of potentially
```
Comment (brtnfld) [inline docs/cd_vals.rst:31]: Did the sentence cut off?

Comment (markcmiller86) [reply docs/cd_vals.rst]: Whoops. Thanks. Fixed now.

**Comment by markcmiller86** [comment]
@brtnfld thanks for the review. My apologies but I had put it initially in as a `ready` PR and then changed to `draft`. It is now again back in the `ready` state.

**Comment by markcmiller86** [comment]
@brtnfld hold off just a min...there is a whole paragraph I still want to add on the behavior of H5Z-ZFP (or any filter) in the presence of partial I/O and chunking.

**Comment by markcmiller86** [comment]
Ok, @brtnfld this is finally ready.

**Review by brtnfld — APPROVED** [review]
LGTM, Jordan also reviewed it, no issues.

**Pull request merged**